### DISCUSSION TITLE
Historical Note: How OVM_ETH works (as of Optimism v0.5.0)

### DISCUSSION THREAD

**Discussion by maurelian** [ethereum-optimism/optimistic-specs#84] [General]
_This is more of an FYI than anything, but I wanted to document this._

Currently:

1. ETH balances on L2 are tracked using `OVM_ETH`, an ERC20 token, at address [`0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000`](https://optimistic.etherscan.io/token/0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000). 
2. How this works is:
	1. When native ETH is transferred or the balance is read, [GETH just looks at the relevant storage](https://github.com/ethereum-optimism/optimism/blob/ff5a87fcc69f15656e8334c253b0a5ecba1dfa1d/l2geth/core/state/statedb.go#L368-L419) slot for `OVM_ETH` the token.
	2. The ERC20 interface itself is uncallable, [all its functions revert](https://github.com/ethereum-optimism/optimism/blob/ff5a87fcc69f15656e8334c253b0a5ecba1dfa1d/packages/contracts/contracts/L2/predeploys/OVM_ETH.sol#L28).
	3. Since `OVM_ETH` inherits from `L2StandardERC20`, this enables it to be minted/burned by the `L2StandardBridge`.

The original motivation for putting L2 ETH in an ERC20 token has to do with the 'containerized' approach behind how the old OVM worked. 
I believe the reason we kept it as an ERC20 on L2 was for backwards compatibility with the old bridge.

**Comment by suahnkim** [comment]
is 0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000 not being used ever? and been replaced with 0x4200000000000000000000000000000000000006 ?
